server/controllers/inputsValidation.ts

```diff
@@ -0,0 +1,101 @@
+export default class InputsValidation {
+  static MAX_CASEREFERENCE_LENGTH = 20
+
+  static validateDateFrom(dateFrom: string) {
+    // optional parameter - if blank, no error
+    if (!dateFrom) {
+      return ''
+    }
+    // Validate that it consists of a day, month and year
+    const [dayString, monthString, yearString] = dateFrom.split('/')
+    if (!dayString || !monthString || !yearString) {
+      return `Date From must be in the format dd/mm/yyyy`
+    }
+    // Validate that day, month and year are all numerical
+    let day
+    let month
+    let year
+    try {
+      day = parseInt(dayString, 10)
+      month = parseInt(monthString, 10)
+      year = parseInt(yearString, 10)
+    } catch (e) {
+      return `Date From must be in the format dd/mm/yyyy`
+    }
+    // Validate that the input date matches the parsed date exactly
+    // This test is useful to catch eg. out of range days/months
+    const inputDateFrom = new Date(year, month - 1, day)
+    if (inputDateFrom.toLocaleDateString('en-gb', { dateStyle: 'short' }) !== dateFrom) {
+      return `Date From must be a valid date`
+    }
+    // Validate that the input date is not in the future
+    if (inputDateFrom > new Date()) {
+      return `Date From must not be in the future`
+    }
+    return ``
+  }
+
+  static validateDateTo(dateTo: string, dateFrom: string) {
+    // optional parameter - if blank, no error
+    if (!dateTo) {
+      return ''
+    }
+    // Validate that it consists of a day, month and year
+    const [dayString, monthString, yearString] = dateTo.split('/')
+    if (!dayString || !monthString || !yearString) {
+      return `Date To must be in the format dd/mm/yyyy`
+    }
+    // Validate that day, month and year are all numerical
+    let day
+    let month
+    let year
+    try {
+      day = parseInt(dayString, 10)
+      month = parseInt(monthString, 10)
+      year = parseInt(yearString, 10)
+    } catch (e) {
+      return `Date To must be in the format dd/mm/yyyy`
+    }
+    // Validate that the input date matches the parsed date exactly
+    // This test is useful to catch eg. out of range days/months
+    const inputDateTo = new Date(year, month - 1, day)
+    if (inputDateTo.toLocaleDateString('en-gb', { dateStyle: 'short' }) !== dateTo) {
+      return `Date To must be a valid date`
+    }
+    // Validate that the input date is not in the future
+    if (inputDateTo > new Date()) {
+      return `Date To must not be in the future`
```

Comment:
Line 5-34 and line  39-68 seems identical but the Date To and Date From, could we simply this into a generic validateDate function?

Reply:
Done with a refactor